pkg/kv/kvserver/allocator_test.go

```diff
@@ -2454,6 +2454,127 @@ var (
 	}
 )
 
+// TestAllocateCandidatesExcludeDrainingNodes checks that draining nodes,
+// as per a store pool's isNodeValidForRoutineReplicaTransfer(),
+// are excluded from the list of candidates for an allocation.
+func TestAllocateCandidatesExcludeDrainingNodes(t *testing.T) {
```

Comment:
nit: there's nothing particular about draining here. You're testing the `isNodeReadyFor....` method. Maybe `TestAllocateCandidatesExcludeNonReadyNodes`.